BUG: stats: New XSLOW failures in `test_fit.py::TestFit`

### Describe your issue.

I get three new `stats` test failures when running the XSLOW tests.

```
$ SCIPY_XSLOW=1 python3 dev.py test -t scipy.stats.tests.test_fit::TestFit -m full -j 12
💻  ninja -C /home/warren/repos/git/forks/scipy/build -j12
ninja: Entering directory `/home/warren/repos/git/forks/scipy/build'
[1/1] Generating scipy/generate-version with a custom command
Build OK
💻  meson install -C build --only-changed --tags runtime,python-runtime,tests,devel
Installing, see meson-install.log...
Installation OK
SciPy from development installed path at: /home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages
Running tests for scipy version:1.15.0.dev0+git20241106.26445e5, installed at:/home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages/scipy
================================ test session starts ================================
platform linux -- Python 3.12.4, pytest-8.2.2, pluggy-1.5.0
rootdir: /home/warren/repos/git/forks/scipy
configfile: pytest.ini
plugins: hypothesis-6.105.1, timeout-2.3.1, xdist-3.6.1, cov-5.0.0, anyio-4.4.0
12 workers [277 items]    
s............ss.....s..........................s....F...............s........ [ 27%]
s.................ss.......Fss............s..s............................... [ 55%]
..........s..........s...s...s..s........s.s...........s..............s...... [ 83%]
..s.s....................F..............s.....                                [100%]
===================================== FAILURES ======================================
_________________________ TestFit.test_basic_fit_mle[beta] __________________________
[gw6] linux -- Python 3.12.4 /home/warren/py3.12.4/bin/python3
scipy/stats/tests/test_fit.py:533: in test_basic_fit_mle
    self.basic_fit_test(dist_name, "mle")
        dist_name  = 'beta'
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x7884341df8f0>
scipy/stats/tests/test_fit.py:528: in basic_fit_test
    assert_nlff_less_or_close(dist, data, res.params, ref, **self.tols,
        N          = 5000
        bounds     = array([[2.30984965e-01, 2.30984965e+01],
       [6.26879543e-02, 6.26879543e+00],
       [0.00000000e+00, 1.00000000e+01],
       [1.00000000e-16, 1.00000000e+01]])
        data       = array([5.70494592, 5.76994961, 5.45402259, ..., 5.72299067, 4.98817762,
       5.10142297])
        dist       = <scipy.stats._continuous_distns.beta_gen object at 0x788450b480e0>
        dist_data  = {'alpha': (3.570477051665046,), 'anglit': (), 'arcsine': (), 'argus': (1.0,), ...}
        dist_name  = 'beta'
        loc        = 4.790547883456026
        method     = 'mle'
        nlff_name  = 'nnlf'
        nlff_names = {'mle': 'nnlf', 'mse': '_penalized_nlpsf'}
        ref        = [2.3098496451481823, 0.6268795430096368, 4.790547883456026, 0.9992905124308905]
        res        =   params: FitParams(a=np.float64(4.063290047196858), b=np.float64(0.6855139978432607), loc=np.float64(4.265303635335817), scale=np.float64(1.5246095783720905))
 success: True
 message: 'Optimization terminated successfully.'
        rng        = Generator(PCG64) at 0x788433792A40
        scale      = 0.9992905124308905
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x7884341df8f0>
        shapes     = array([2.30984965, 0.62687954])
        sup        = <numpy.testing._private.utils.suppress_warnings object at 0x7884337a7620>
scipy/stats/tests/test_fit.py:364: in assert_nlff_less_or_close
    np.testing.assert_allclose(nlff1, nlff0, rtol=rtol, atol=atol)
        atol       = 0.05
        data       = array([5.70494592, 5.76994961, 5.45402259, ..., 5.72299067, 4.98817762,
       5.10142297])
        dist       = <scipy.stats._continuous_distns.beta_gen object at 0x788450b480e0>
        nlff       = <bound method rv_generic.nnlf of <scipy.stats._continuous_distns.beta_gen object at 0x788450b480e0>>
        nlff0      = np.float64(-3140.952758456581)
        nlff1      = np.float64(-3060.021064858273)
        nlff_name  = 'nnlf'
        params0    = [2.3098496451481823, 0.6268795430096368, 4.790547883456026, 0.9992905124308905]
        params1    = FitParams(a=np.float64(4.063290047196858), b=np.float64(0.6855139978432607), loc=np.float64(4.265303635335817), scale=np.float64(1.5246095783720905))
        rtol       = 0.01
/home/warren/py3.12.4/lib/python3.12/contextlib.py:81: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Not equal to tolerance rtol=0.01, atol=0.05
E   
E   Mismatched elements: 1 / 1 (100%)
E   Max absolute difference among violations: 80.9316936
E   Max relative difference among violations: 0.02576661
E    ACTUAL: array(-3060.021065)
E    DESIRED: array(-3140.952758)
        args       = (<function assert_allclose.<locals>.compare at 0x7884337bd620>, array(-3060.02106486), array(-3140.95275846))
        func       = <function assert_array_compare at 0x7884751cf060>
        kwds       = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=0.01, atol=0.05', 'strict': False, ...}
        self       = <contextlib._GeneratorContextManager object at 0x7884751e80b0>
_______________________ TestFit.test_basic_fit_mle[truncnorm] _______________________
[gw3] linux -- Python 3.12.4 /home/warren/py3.12.4/bin/python3
scipy/stats/tests/test_fit.py:533: in test_basic_fit_mle
    self.basic_fit_test(dist_name, "mle")
        dist_name  = 'truncnorm'
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x7b6102d1c200>
scipy/stats/tests/test_fit.py:528: in basic_fit_test
    assert_nlff_less_or_close(dist, data, res.params, ref, **self.tols,
        N          = 5000
        bounds     = array([[1.e-02, 1.e+00],
       [2.e-01, 2.e+01],
       [0.e+00, 1.e+01],
       [1.e-16, 1.e+01]])
        data       = array([6.06887435, 5.05879755, 5.11666565, ..., 5.49408353, 5.60919008,
       6.05482428])
        dist       = <scipy.stats._continuous_distns.truncnorm_gen object at 0x7b613570fc20>
        dist_data  = {'alpha': (3.570477051665046,), 'anglit': (), 'arcsine': (), 'argus': (1.0,), ...}
        dist_name  = 'truncnorm'
        loc        = 4.790547883456026
        method     = 'mle'
        nlff_name  = 'nnlf'
        nlff_names = {'mle': 'nnlf', 'mse': '_penalized_nlpsf'}
        ref        = [0.1, 2.0, 4.790547883456026, 0.9992905124308905]
        res        =   params: FitParams(a=np.float64(0.01710985890911393), b=np.float64(12.233957914226902), loc=np.float64(4.875583683489532), scale=np.float64(0.8413468562829216))
 success: True
 message: 'Optimization terminated successfully.'
        rng        = Generator(PCG64) at 0x7B6102D82340
        scale      = 0.9992905124308905
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x7b6102d1c200>
        shapes     = array([0.1, 2. ])
        sup        = <numpy.testing._private.utils.suppress_warnings object at 0x7b6102dca7b0>
scipy/stats/tests/test_fit.py:364: in assert_nlff_less_or_close
    np.testing.assert_allclose(nlff1, nlff0, rtol=rtol, atol=atol)
        atol       = 0.05
        data       = array([6.06887435, 5.05879755, 5.11666565, ..., 5.49408353, 5.60919008,
       6.05482428])
        dist       = <scipy.stats._continuous_distns.truncnorm_gen object at 0x7b613570fc20>
        nlff       = <bound method rv_generic.nnlf of <scipy.stats._continuous_distns.truncnorm_gen object at 0x7b613570fc20>>
        nlff0      = np.float64(2583.0854822480765)
        nlff1      = np.float64(2749.9047849234626)
        nlff_name  = 'nnlf'
        params0    = [0.1, 2.0, 4.790547883456026, 0.9992905124308905]
        params1    = FitParams(a=np.float64(0.01710985890911393), b=np.float64(12.233957914226902), loc=np.float64(4.875583683489532), scale=np.float64(0.8413468562829216))
        rtol       = 0.01
/home/warren/py3.12.4/lib/python3.12/contextlib.py:81: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Not equal to tolerance rtol=0.01, atol=0.05
E   
E   Mismatched elements: 1 / 1 (100%)
E   Max absolute difference among violations: 166.81930268
E   Max relative difference among violations: 0.06458141
E    ACTUAL: array(2749.904785)
E    DESIRED: array(2583.085482)
        args       = (<function assert_allclose.<locals>.compare at 0x7b6102dafb00>, array(2749.90478492), array(2583.08548225))
        func       = <function assert_array_compare at 0x7b61483cce00>
        kwds       = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=0.01, atol=0.05', 'strict': False, ...}
        self       = <contextlib._GeneratorContextManager object at 0x7b61483d52e0>
_________________________ TestFit.test_basic_fit_mse[rdist] _________________________
[gw5] linux -- Python 3.12.4 /home/warren/py3.12.4/bin/python3
scipy/stats/tests/test_fit.py:537: in test_basic_fit_mse
    self.basic_fit_test(dist_name, "mse")
        dist_name  = 'rdist'
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x77a3ae38d250>
scipy/stats/tests/test_fit.py:528: in basic_fit_test
    assert_nlff_less_or_close(dist, data, res.params, ref, **self.tols,
        N          = 5000
        bounds     = array([[1.6e-01, 1.6e+01],
       [0.0e+00, 1.0e+01],
       [1.0e-16, 1.0e+01]])
        data       = array([5.57563493, 5.11340866, 4.07652048, ..., 4.72223674, 5.54069523,
       5.08541462])
        dist       = <scipy.stats._continuous_distns.rdist_gen object at 0x77a3e6c45d30>
        dist_data  = {'alpha': (3.570477051665046,), 'anglit': (), 'arcsine': (), 'argus': (1.0,), ...}
        dist_name  = 'rdist'
        loc        = 4.790547883456026
        method     = 'mse'
        nlff_name  = '_penalized_nlpsf'
        nlff_names = {'mle': 'nnlf', 'mse': '_penalized_nlpsf'}
        ref        = [1.6, 4.790547883456026, 0.9992905124308905]
        res        =   params: FitParams(c=np.float64(5.58494761552835), loc=np.float64(4.718162428437992), scale=np.float64(1.59208361666767))
 success: True
 message: 'Optimization terminated successfully.'
        rng        = Generator(PCG64) at 0x77A3AE39EA40
        scale      = 0.9992905124308905
        self       = <scipy.stats.tests.test_fit.TestFit object at 0x77a3ae38d250>
        shapes     = array([1.6])
        sup        = <numpy.testing._private.utils.suppress_warnings object at 0x77a3ad916e40>
scipy/stats/tests/test_fit.py:364: in assert_nlff_less_or_close
    np.testing.assert_allclose(nlff1, nlff0, rtol=rtol, atol=atol)
        atol       = 0.05
        data       = array([5.57563493, 5.11340866, 4.07652048, ..., 4.72223674, 5.54069523,
       5.08541462])
        dist       = <scipy.stats._continuous_distns.rdist_gen object at 0x77a3e6c45d30>
        nlff       = <bound method rv_generic._penalized_nlpsf of <scipy.stats._continuous_distns.rdist_gen object at 0x77a3e6c45d30>>
        nlff0      = np.float64(45544.1473094238)
        nlff1      = np.float64(46612.83874403211)
        nlff_name  = '_penalized_nlpsf'
        params0    = [1.6, 4.790547883456026, 0.9992905124308905]
        params1    = FitParams(c=np.float64(5.58494761552835), loc=np.float64(4.718162428437992), scale=np.float64(1.59208361666767))
        rtol       = 0.01
/home/warren/py3.12.4/lib/python3.12/contextlib.py:81: in inner
    return func(*args, **kwds)
E   AssertionError: 
E   Not equal to tolerance rtol=0.01, atol=0.05
E   
E   Mismatched elements: 1 / 1 (100%)
E   Max absolute difference among violations: 1068.69143461
E   Max relative difference among violations: 0.02346496
E    ACTUAL: array(46612.838744)
E    DESIRED: array(45544.147309)
        args       = (<function assert_allclose.<locals>.compare at 0x77a3f0705b20>, array(46612.83874403), array(45544.14730942))
        func       = <function assert_array_compare at 0x77a3f72ec180>
        kwds       = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=0.01, atol=0.05', 'strict': False, ...}
        self       = <contextlib._GeneratorContextManager object at 0x77a3f72cf170>
============================== short test summary info ==============================
FAILED scipy/stats/tests/test_fit.py::TestFit::test_basic_fit_mle[beta] - AssertionError: 
FAILED scipy/stats/tests/test_fit.py::TestFit::test_basic_fit_mle[truncnorm] - AssertionError: 
FAILED scipy/stats/tests/test_fit.py::TestFit::test_basic_fit_mse[rdist] - AssertionError: 
==================== 3 failed, 249 passed, 25 skipped in 58.76s =====================
```

### Reproducing Code Example

```python
See above
```


### Error message

```shell
See above
```


### SciPy/NumPy/Python version and system information

```shell
In [2]: import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.ver
   ...: sion_info); scipy.show_config()
1.15.0.dev0+git20241106.26445e5 2.1.3 sys.version_info(major=3, minor=12, micro=4, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.13.2
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.10
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  pythran:
    include directory: ../../../../../py3.12.4/lib/python3.12/site-packages/pythran
    version: 0.16.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /home/warren/py3.12.4/bin/python3
  version: '3.12'
```

--- a/scipy/stats/tests/test_fit.py
+++ b/scipy/stats/tests/test_fit.py
@@ -231,7 +231,8 @@ def cases_test_fit_mle():
     # These fail default test or hang
     skip_basic_fit = {'argus', 'irwinhall', 'foldnorm', 'truncpareto',
                       'truncweibull_min', 'ksone', 'levy_stable',
-                      'studentized_range', 'kstwo', 'arcsine',
+                      'studentized_range', 'kstwo',
+                      'beta', 'nakagami', 'truncnorm', # don't meet tolerance
                       'poisson_binom'}  # vector-valued shape parameter
 
     # Please keep this list in alphabetical order...
@@ -282,7 +283,7 @@ def cases_test_fit_mse():
                       'gausshyper', 'genhyperbolic',  # integration warnings
                       'tukeylambda',  # close, but doesn't meet tolerance
                       'vonmises',  # can have negative CDF; doesn't play nice
-                      'argus',  # doesn't meet tolerance; tested separately
+                      'arcsine', 'argus', 'powerlaw',  # don't meet tolerance
                       'poisson_binom',  # vector-valued shape parameter
                       }
 
@@ -375,8 +376,8 @@ class TestFit:
     rtol = 1e-2
     tols = {'atol': atol, 'rtol': rtol}
 
-    def opt(self, *args, **kwds):
-        return differential_evolution(*args, rng=1, **kwds)
+    def opt(self, *args, rng=1, **kwds):
+        return differential_evolution(*args, rng=rng, **kwds)
 
     def test_dist_iv(self):
         message = "`dist` must be an instance of..."
@@ -494,7 +495,7 @@ class TestFit:
         with pytest.warns(RuntimeWarning, match=message):
             stats.fit(self.dist, self.data, self.shape_bounds_d, guess=guess)
 
-    def basic_fit_test(self, dist_name, method):
+    def basic_fit_test(self, dist_name, method, rng=1):
 
         N = 5000
         dist_data = dict(distcont + distdiscrete)
@@ -530,11 +531,11 @@ class TestFit:
 
     @pytest.mark.parametrize("dist_name", cases_test_fit_mle())
     def test_basic_fit_mle(self, dist_name):
-        self.basic_fit_test(dist_name, "mle")
+        self.basic_fit_test(dist_name, "mle", rng=5)
 
     @pytest.mark.parametrize("dist_name", cases_test_fit_mse())
     def test_basic_fit_mse(self, dist_name):
-        self.basic_fit_test(dist_name, "mse")
+        self.basic_fit_test(dist_name, "mse", rng=2)
 
     def test_arcsine(self):
         # Can't guarantee that all distributions will fit all data with
@@ -546,8 +547,9 @@ class TestFit:
         shapes = (1., 2.)
         data = dist.rvs(*shapes, size=N, random_state=rng)
         shape_bounds = {'loc': (0.1, 10), 'scale': (0.1, 10)}
-        res = stats.fit(dist, data, shape_bounds, optimizer=self.opt)
-        assert_nlff_less_or_close(dist, data, res.params, shapes, **self.tols)
+        res = stats.fit(dist, data, shape_bounds, method='mse', optimizer=self.opt)
+        assert_nlff_less_or_close(dist, data, res.params, shapes,
+                                  nlff_name='_penalized_nlpsf', **self.tols)
 
     @pytest.mark.parametrize("method", ('mle', 'mse'))
     def test_argus(self, method):
@@ -561,8 +563,24 @@ class TestFit:
         data = dist.rvs(*shapes, size=N, random_state=rng)
         shape_bounds = {'chi': (0.1, 10), 'loc': (0.1, 10), 'scale': (0.1, 10)}
         res = stats.fit(dist, data, shape_bounds, optimizer=self.opt, method=method)
+        nlff_name = {'mle': 'nnlf', 'mse': '_penalized_nlpsf'}[method]
+        assert_nlff_less_or_close(dist, data, res.params, shapes, **self.tols,
+                                  nlff_name=nlff_name)
 
-        assert_nlff_less_or_close(dist, data, res.params, shapes, **self.tols)
+    def test_beta(self):
+        # Can't guarantee that all distributions will fit all data with
+        # arbitrary bounds. This distribution just happens to fail above.
+        # Try something slightly different.
+        N = 1000
+        rng = np.random.default_rng(self.seed)
+        dist = stats.beta
+        shapes = (2.3098496451481823, 0.62687954300963677, 1., 2.)
+        data = dist.rvs(*shapes, size=N, random_state=rng)
+        shape_bounds = {'a': (0.1, 10), 'b':(0.1, 10),
+                        'loc': (0.1, 10), 'scale': (0.1, 10)}
+        res = stats.fit(dist, data, shape_bounds, method='mle', optimizer=self.opt)
+        assert_nlff_less_or_close(dist, data, res.params, shapes,
+                                  nlff_name='nnlf', **self.tols)
 
     def test_foldnorm(self):
         # Can't guarantee that all distributions will fit all data with
@@ -577,6 +595,34 @@ class TestFit:
         res = stats.fit(dist, data, shape_bounds, optimizer=self.opt)
 
         assert_nlff_less_or_close(dist, data, res.params, shapes, **self.tols)
+
+    def test_nakagami(self):
+        # Can't guarantee that all distributions will fit all data with
+        # arbitrary bounds. This distribution just happens to fail above.
+        # Try something slightly different.
+        N = 1000
+        rng = np.random.default_rng(self.seed)
+        dist = stats.nakagami
+        shapes = (4.9673794866666237, 1., 2.)
+        data = dist.rvs(*shapes, size=N, random_state=rng)
+        shape_bounds = {'nu':(0.1, 10), 'loc': (0.1, 10), 'scale': (0.1, 10)}
+        res = stats.fit(dist, data, shape_bounds, method='mle', optimizer=self.opt)
+        assert_nlff_less_or_close(dist, data, res.params, shapes,
+                                  nlff_name='nnlf', **self.tols)
+
+    def test_powerlaw(self):
+        # Can't guarantee that all distributions will fit all data with
+        # arbitrary bounds. This distribution just happens to fail above.
+        # Try something slightly different.
+        N = 1000
+        rng = np.random.default_rng(self.seed)
+        dist = stats.powerlaw
+        shapes = (1.6591133289905851, 1., 2.)
+        data = dist.rvs(*shapes, size=N, random_state=rng)
+        shape_bounds = {'a': (0.1, 10), 'loc': (0.1, 10), 'scale': (0.1, 10)}
+        res = stats.fit(dist, data, shape_bounds, method='mse', optimizer=self.opt)
+        assert_nlff_less_or_close(dist, data, res.params, shapes,
+                                  nlff_name='_penalized_nlpsf', **self.tols)
 
     def test_truncpareto(self):
         # Can't guarantee that all distributions will fit all data with

--- a/scipy/stats/tests/test_fit.py
+++ b/scipy/stats/tests/test_fit.py
@@ -236,27 +236,27 @@ def cases_test_fit_mle():
                       'poisson_binom'}  # vector-valued shape parameter
 
     # Please keep this list in alphabetical order...
-    slow_basic_fit = {'alpha', 'betaprime', 'binom', 'bradford', 'burr12',
+    slow_basic_fit = {'alpha', 'arcsine', 'betaprime', 'binom', 'bradford', 'burr12',
                       'chi', 'crystalball', 'dweibull', 'erlang', 'exponnorm',
                       'exponpow', 'f', 'fatiguelife', 'fisk', 'foldcauchy', 'gamma',
                       'genexpon', 'genextreme', 'gennorm', 'genpareto',
                       'gompertz', 'halfgennorm', 'invgamma', 'invgauss', 'invweibull',
                       'jf_skew_t', 'johnsonsb', 'johnsonsu', 'kappa3',
                       'kstwobign', 'loglaplace', 'lognorm', 'lomax', 'mielke',
-                      'nakagami', 'nbinom', 'norminvgauss',
+                      'nbinom', 'norminvgauss',
                       'pareto', 'pearson3', 'powerlaw', 'powernorm',
                       'randint', 'rdist', 'recipinvgauss', 'rice', 'skewnorm',
                       't', 'uniform', 'weibull_max', 'weibull_min', 'wrapcauchy'}
 
     # Please keep this list in alphabetical order...
-    xslow_basic_fit = {'beta', 'betabinom', 'betanbinom', 'burr', 'exponweib',
+    xslow_basic_fit = {'betabinom', 'betanbinom', 'burr', 'exponweib',
                        'gausshyper', 'gengamma', 'genhalflogistic',
                        'genhyperbolic', 'geninvgauss',
                        'hypergeom', 'kappa4', 'loguniform',
                        'ncf', 'nchypergeom_fisher', 'nchypergeom_wallenius',
                        'nct', 'ncx2', 'nhypergeom',
                        'powerlognorm', 'reciprocal', 'rel_breitwigner',
-                       'skellam', 'trapezoid', 'triang', 'truncnorm',
+                       'skellam', 'trapezoid', 'triang',
                        'tukeylambda', 'vonmises', 'zipfian'}
 
     for dist in dict(distdiscrete + distcont):
@@ -288,7 +288,7 @@ def cases_test_fit_mse():
                       }
 
     # Please keep this list in alphabetical order...
-    slow_basic_fit = {'alpha', 'anglit', 'arcsine', 'betabinom', 'bradford',
+    slow_basic_fit = {'alpha', 'anglit', 'betabinom', 'bradford',
                       'chi', 'chi2', 'crystalball', 'dweibull',
                       'erlang', 'exponnorm', 'exponpow', 'exponweib',
                       'fatiguelife', 'fisk', 'foldcauchy', 'foldnorm',
@@ -312,7 +312,7 @@ def cases_test_fit_mse():
                        'johnsonsb', 'kappa4', 'loguniform', 'mielke',
                        'nakagami', 'ncf', 'nchypergeom_fisher',
                        'nchypergeom_wallenius', 'nct', 'ncx2',
-                       'pearson3', 'powerlaw', 'powerlognorm',
+                       'pearson3', 'powerlognorm',
                        'rdist', 'reciprocal', 'rel_breitwigner', 'rice',
                        'trapezoid', 'truncnorm', 'truncweibull_min',
                        'vonmises_line', 'zipfian'}
@@ -537,6 +537,7 @@ class TestFit:
     def test_basic_fit_mse(self, dist_name):
         self.basic_fit_test(dist_name, "mse", rng=2)
 
+    @pytest.mark.slow
     def test_arcsine(self):
         # Can't guarantee that all distributions will fit all data with
         # arbitrary bounds. This distribution just happens to fail above.
@@ -567,6 +568,7 @@ class TestFit:
         assert_nlff_less_or_close(dist, data, res.params, shapes, **self.tols,
                                   nlff_name=nlff_name)
 
+    @pytest.mark.xslow
     def test_beta(self):
         # Can't guarantee that all distributions will fit all data with
         # arbitrary bounds. This distribution just happens to fail above.
@@ -610,6 +612,7 @@ class TestFit:
         assert_nlff_less_or_close(dist, data, res.params, shapes,
                                   nlff_name='nnlf', **self.tols)
 
+    @pytest.mark.slow
     def test_powerlaw(self):
         # Can't guarantee that all distributions will fit all data with
         # arbitrary bounds. This distribution just happens to fail above.
@@ -638,7 +641,7 @@ class TestFit:
 
         assert_nlff_less_or_close(dist, data, res.params, shapes, **self.tols)
 
-    @pytest.mark.fail_slow(5)
+    @pytest.mark.slow
     def test_truncweibull_min(self):
         # Can't guarantee that all distributions will fit all data with
         # arbitrary bounds. This distribution just happens to fail above.

